redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -0,0 +1,260 @@
+                                            .jsonValue["Oem"]["IBM"]
+                                                      ["InputPowerHistoryItem"]
+                                                      ["Max"];
+                                    for (const auto& values : *values_ptr)
+                                    {
+                                        // The first value returned is the
+                                        // timestamp, already handled in
+                                        // getAverageValues().
+                                        // The second value returned is in
+                                        // watts. It is the maximum watts this
+                                        // power supply has used in a 30 second
+                                        // interval.
+                                        maximums.push_back(std::get<1>(values));
+                                    }
+                                }
+                                else
+                                {
+                                    BMCWEB_LOG_ERROR
+                                        << "Failed to find power supply input "
+                                           "history Values data for:"
+                                        << powerSupplyID;
+                                }
+                            },
+                            serviceName, maximumPath,
+                            "org.freedesktop.DBus.Properties", "Get", interface,
+                            "Values");
+                    }
+                }
+            }
+        },
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject", maximumPath,
+        std::array<const char*, 1>{
+            "org.open_power.Sensor.Aggregation.History.Maximum"});
+}
+
+/**
+ * Systems derived class for delivering OemPowerSupplyMetric Schema.
+ */
+inline void requestRoutesPowerSupplyMetrics(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Chassis/<str>/PowerSubsystem/"
+                      "PowerSupplies/<str>/Metrics")
+        .privileges({{"Login"}})
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& param, const std::string& param2) {
+                const std::string& chassisId = param;
+                const std::string& powerSupplyId = param2;
+
+                BMCWEB_LOG_INFO << "ChassisID: " << chassisId;
+                BMCWEB_LOG_INFO << "PowerSupplyID: " << powerSupplyId;
+
+                asyncResp->res.jsonValue["@odata.type"] =
+                    "#OemPowerSupplyMetric.v1_0_0";
+                asyncResp->res.jsonValue["@odata.id"] =
+                    "/redfish/v1/Chassis/" + chassisId +
+                    "/PowerSubSystem/PowerSupplies/" + powerSupplyId +
```

Comment:
Did you run the validator?
Nope, not yet... what could possibly go wrong?
```
*** /redfish/v1/Chassis/chassis/PowerSubsystem/PowerSupplies/powersupply0/Metrics
/redfish/v1/Chassis/chassis/PowerSubsystem/PowerSupplies/powersupply0/Metrics @odata.id: Expected @odata.id to match URI link /redfish/v1/Chassis/chassis/PowerSubSystem/PowerSupplies/powersupply0/Metrics
Unable to gather property info for URI /redfish/v1/Chassis/chassis/PowerSubsystem/PowerSupplies/powersupply0/Metrics: MissingSchemaError('Could not find any Schema with these parameters OemPowerSupplyMetric')
```
Oh, okay.